src/main/scala/io/iohk/ethereum/domain/BlockHeader.scala

```diff
@@ -79,11 +82,32 @@ case class BlockHeader(
 
 object BlockHeader {
 
+  val NumOfFields = 16
```

Comment:
If you want to be fully general:

```suggestion
  val NumOfFields = {
    val bEmpty = akka.util.ByteString.empty
    val header = BlockHeader(
      bEmpty, bEmpty, bEmpty, bEmpty, bEmpty, bEmpty, bEmpty,
      0,0,0,0,
      0L,
      bEmpty, bEmpty, bEmpty)
    header.productArity // 16
 }
```